errors.go

```diff
@@ -100,6 +100,27 @@ type TimeoutError struct {
 	Err error
 }
 
+// HandshakeError indicates that the handshake failed.
+type HandshakeError struct {
```

Comment:
HandshakeError should provide `Temporary()` and `Timeout()` which transparently returns a value from underlying error (`Err.Temporary()` and `Err.Timeout()`) or `false` if the error doesn't implement it.

Reply:
done